### DISCUSSION TITLE
Big list of (mostly) small QOL features that could be added.

### DISCUSSION THREAD

**Discussion by Captain-Quack** [codewars/codewars.com#2888] [Ideas]
- ~~Instead of sorting by Low Satisfaction, change it to High Satisfaction. The bad katas should not be at the top of a sorting option.~~
- Clicking on tags while browsing kata's allows you to filter the selected category quickly, but not in the kata description. Have it filter searches there as well?
- Change "Featured Tags" to the featured weekly kata on Twitter. 
- *Please* reduce the contrast in light mode a bit: it's tough to look at!
- Solutions from (clan members/allies/followed users) could show at the top when viewing the answers after a kata. 
- "This comment is hidden because it contains spoiler information about the solution" is a bit pointless. Consider hiding the comment entirely?
- Have the option to delete sub-comments, but replace them with "This comment has been deleted." The comment could still be visible to existing contributors to the conversation.
- Ability to sort the kata in the profile. For example, you could sort the "Unfinished Kata" by difficulty, attempts made, etc.
- Move the "Translations" (from the profile) to a sub-category of Authored Kata. 
- Include the same code set up from the example for Kata creation in Kumites and Translation. Not having this is a pain for languages with numerous boilerplate lines, like COBOL and C#.
- Can the user info bar still be visible when reading the docs? I realize this might not be possible with Docasaurus, but it would be nice to add if so. 
- Referrals are a really small portion of received honor points, and I don't think it should take up a whole bar in Honor Breakdown.
- Have some more UI color options? A dark blue or sepia theme might look nice.
- Weight the default kata sorting based on user rank. 

Thanks for reading, and Happy New Year!

**Comment by hobovsky** [comment]
Ad _"Instead of sorting by Low Satisfaction, change it to High Satisfaction. The bad katas should not be at the top of a sorting option."_ - this sorting option has been added _specifically_ to make it easy to find bad kata, to support people willing to fix and/or remove bad kata.

Sorting by satisfaction tends to be terribly distorted by kata which have 0, 1, or 2 votes: for "Sort by: low satisfaction", over two top pages is taken by kata with "0% satisfaction, calculated from 0 votes". I believe that "Sort by: high satisfaction" would be similarly skewed by kata with "100% satisfaction, calculated from 1 vote". I _think_ that "sort by: positive feedback" is meant to account for this somehow, but I never use it so I don't know if it works properly.

Anyway, having the "sort by: high satisfaction" option available or not, the option to sort by low satisfaction is there on purpose, to support content maintenance activities.

**Comment by Captain-Quack** [comment]
Oh, I see.

I think the positive feedback sorts by "Very Satisfied" votes, which the top kata ([Multiples of 3 or 5](https://www.codewars.com/kata/514b92a657cdc65150000006)) has 25000 of.

**Comment by Kacarott** [comment]
> Please reduce the contrast in light mode a bit: it's tough to look at!

I definitely agree, not that I use light mode anyway.

> Solutions from (clan members/allies/followed users) could show at the top when viewing the answers after a kata.

I disagree with this. It is very easy to filter for solutions of users you follow, but often the first thing I want to see is the "top rated" solution to see if there is some elegant solution I missed. Perhaps it might be nice to set the default solution sorting in user profile?

> "This comment is hidden because it contains spoiler information about the solution" is a bit pointless. Consider hiding the comment entirely?

This seems like it would be very confusing, and wouldn't provide any benefit besides slightly reducing clutter for people who haven't solved yet.

Many of your other minor suggestions seem nice.

**Comment by kazk** [comment]
Please create separate discussions. This is too difficult to discuss/vote/discover and to keep track of.

**Comment by iris-is-me** [comment]
---

I have another Qol for clan management

- [ ] admin panel for clan owners  
- [ ] invite system in the UI  
- [ ] kick / ban button
- [ ] clan member privileges/rankings (kinda like discord mod & admins)
- [ ] a way to make a clan private or smth (if an invite system was added then this might not be needed)
